modules/modules.nix

```diff
@@ -74,6 +74,7 @@ let
     config._module.args.baseModules = modules;
     config._module.args.pkgs = lib.mkDefault pkgs;
     config._module.check = check;
+    config.lib.dag = import lib/dag.nix { inherit lib; };
```

Comment:
I think we could use this opportunity to make it possible to add more general purpose functions to the library. Adding a `lib/default.nix` with
```nix
{ lib }: {
  dag = import ./dag { inherit lib; };
  ...
}
```

Then `config.lib.home = import ./lib { inherit lib; };`

This also calls for a DAG library refactor, such that you can use `config.lib.dag.entryAfter` instead of `config.lib.dag.dagEntryAfter`, etc.

Reply:
Yeah, I was thinking about the function renaming. Unfortunately it would break the configuration of anybody who uses the dag functions directly.

I'm now thinking the `config.lib.dag` entries actually should be different from the `dag.nix` file. I.e., to have `config.lib.dag = { …; entryAfter = dagEntryAfter; … };` etc. The old `dag.nix` could be deprecated and removed in the future.

Also agree on the use of `lib/default.nix`.

Reply:
Good idea regarding the slow deprecation